lms/djangoapps/instructor_task/tests/test_tasks_helper.py

```diff
@@ -538,3 +543,182 @@ def test_move_users_to_same_cohort(self):
+        """
+        user_profile = UserFactory(username=user.username, email=user.email).profile
+        user_profile.allow_certificate = not is_embargoed
+        user_profile.save()
+
+    def set_user_verification_status(self, user, status):
+        """
+        Set a users verification status.
+        """
+        attempt, _ = SoftwareSecurePhotoVerification.objects.get_or_create(user=user)
+        attempt.status = status
+        attempt.save()
+
+    def _verify_csv_certificate_data(self, username, expected_data):
+        """
+        Verify grade report data.
+        """
+        with patch('instructor_task.tasks_helper._get_current_task'):
+            upload_grades_csv(None, None, self.course.id, None, 'graded')
+            report_store = ReportStore.from_config()
+            report_csv_filename = report_store.links_for(self.course.id)[0][0]
+            with open(report_store.path_to(self.course.id, report_csv_filename)) as csv_file:
+                for row in unicodecsv.DictReader(csv_file):
+                    if row.get('username') == username:
+                        csv_row_data = [row[column] for column in self.csv_cert_info_header]
+                        self.assertEqual(csv_row_data, expected_data)
+
+    def _create_user_data(self,
+                          user_enroll_mode,
+                          has_passed,
+                          whitelisted,
+                          is_embargoed,
+                          verification_status,
+                          certificate_status,
+                          certificate_mode,
+                          expected_data):
+        """
+        Create user data to be used during grade report generation.
+        """
+
+        self.create_student('u1', mode=user_enroll_mode)
+        user = User.objects.get(username='u1')
+
+        if has_passed:
+            self.submit_student_answer('u1', 'test_problem', ['choice_1'])
+
+        self.user_in_whitelist(user, self.course.id, whitelisted)
+
+        self.user_is_embargoed(user, is_embargoed)
+
+        if user_enroll_mode in CourseMode.VERIFIED_MODES:
+            self.set_user_verification_status(user, verification_status)
+
+        self.generate_certificate(user, self.course.id, certificate_status, certificate_mode)
+
+        # Verify columns values in grade report
+        self._verify_csv_certificate_data(user.username, expected_data)
+
+    def test_failed_non_whitelist_user_certificate_info(self):
+        """
+        Verify certificate info for failed user who is not in whitelist.
+        """
+        self._create_user_data(
+            user_enroll_mode='verified',
+            has_passed=False,
+            whitelisted=False,
+            is_embargoed=False,
+            verification_status='approved',
+            certificate_status='notpassing',
+            certificate_mode='honor',
+            expected_data=['verified', 'ID Verified', 'N', 'N', 'N/A']
+        )
```

Comment:
You can break it into multiple lines like this: https://github.com/symbolist/db-utils/blob/master/db_utils/tests/test_transaction.py#L89